### PR TITLE
Fix issue #2867: Honor the default unit from the nutrient taxonomy

### PR DESCRIPTION
**Pull request by machiav3lli**
####  Description
A first Iteration of our suggested solution in the discussion under #2867 .
For this one we focused on the logic managing binding each nutrient with a default value and this saved statically in a map.
A smal test is also included that checks if the map still binds the true default values for some 4 cherry-picked nutrients.

#### Related issues
#2867. This solves the issue fully, but another variant is also possible. The two variants are:
- Use a static map to bind (like in this first iteration)
    - Have to be maintained (Should be in the future new nutrients to be considered).
    - More efficient considering offline usage of the app.
- Use OFF JSON data to get the default value of each nutrient
    - Gets maintained on the server side.
    - Less efficient considering calls etc.

We would be interested to hear the opinion of the community+maintainer(s)…

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/edit/nutrition/ProductEditNutritionFactsFragment.kt [hunk 1]
```diff
@@ -712,17 +711,25 @@ class ProductEditNutritionFactsFragment : ProductEditFragment() {
         }
 
     private fun displayAddNutrientDialog() {
-        val nutrients = resources.getStringArray(R.array.nutrients_array)
+        val nutrientsDefUnits = resources.getStringArray(R.array.nutrients_array).zip(PARAMS_OTHER_NUTRIENTS_DEFAULT_UNITS.keys)
+        val filteredNutrients = resources.getStringArray(R.array.nutrients_array)
             .filterIndexed { index, _ -> index !in usedNutrientsIndexes }
-            .sortedWith(Collator.getInstance(Locale.getDefault()))
+            .sortedBy { it.lowercase() }
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/edit/nutrition/ProductEditNutritionFactsFragment.kt]: Why did you remove this? By using `sortedBy { it.lowercase() }` we're not considering all Unicode languages.

Comment (machiav3lli) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/edit/nutrition/ProductEditNutritionFactsFragment.kt]: Sorry, was a part of one of a previous implementation idea that we forgot to clean up. But now it's back

**Comment by VaiTon** [comment]
@machiav3lli tests are failing

**Comment by machiav3lli** [comment]
@VaiTon yeah, I've forgotten the import

**Comment by VaiTon** [comment]
@machiav3lli could you fix it so that tests work?

**Comment by machiav3lli** [comment]
> @machiav3lli could you fix it so that tests work?

I've already fixed before I wrote the note about the missing import and it builds just fine on my side. The UnitTest that is failing is:
_openfoodfacts.github.scrachx.openfood.network.ProductsAPITest > productByAdditive_e301_productsFound_
```java
@Test
    fun productByAdditive_e301_productsFound() {
        val fieldsToFetchFacets = "brands,product_name,image_small_url,quantity,nutrition_grades_tags"
        val response = prodClient.getProductsByAdditive("e301-sodium-ascorbate", fieldsToFetchFacets).blockingGet() as Search
        assertThat(response).hasFoundProducts()
    }
```

 I guess it could be an error/change on the server side.

Lint OFF seems also to be failing, but all the mentioned errors doesn't relate to the classes touched by this PR in anyway and for the most part are (string) formatting issues.

Please check those and tell me if we should try to fix those or if it's maybe someone else's responsibility (server for UnitTest, OFF-maintainer for string formats).
